src/handler.rs

```diff
@@ -535,20 +548,193 @@ where
 
     fn get_inflight_fd(
         &mut self,
-        _inflight: &vhost::vhost_user::message::VhostUserInflight,
+        inflight: &vhost::vhost_user::message::VhostUserInflight,
     ) -> VhostUserResult<(vhost::vhost_user::message::VhostUserInflight, File)> {
-        // Assume the backend hasn't negotiated the inflight feature; it
-        // wouldn't be correct for the backend to do so, as we don't (yet)
-        // provide a way for it to handle such requests.
-        Err(VhostUserError::InvalidOperation)
+        let ret_val = -1;
+        // Total size of the inflight queue region
+        let mut total_mmap_size =
+            self.get_inflight_queue_size(inflight.queue_size) * inflight.num_queues as usize;
+
+        // Create a memfd region to hold the queues for inflight I/O tracking
+        let mut inflight_fd = -1;
+        let mmap_ptr = self.memfd_alloc(total_mmap_size, &mut inflight_fd)?;
```

Comment:
It's more straight-forward to make memfd_alloc() to return `(mmap_ptr, inflight_file)`.